Sources/Plasma/FeatureLib/pfConsole/pfConsole.cpp

```diff
@@ -1002,6 +953,120 @@ void    pfConsole::IUpdateTooltip()
     }
 }
 
+//// IExecuteWorkingLine /////////////////////////////////////////////////////
+
+void    pfConsole::IExecuteWorkingLine()
+{
+    int   i, eol;
+    char* c;
+
+    // leave leading space for Python multi lines (need the indents!)
+    if (fPythonMultiLines == 0) {
+        // Clean up working line by removing any leading whitespace
+        for (c = fWorkingLine; *c == ' ' || *c == '\t'; c++);
+        for (i = 0; *c != 0; i++, c++)
+            fWorkingLine[i] = *c;
+        fWorkingLine[i] = 0;
+        eol = i;
+    }
+
+    if (fWorkingLine[0] == 0 && !fHelpMode && !fPythonMode) {
+        // Blank line--just print a blank line to the console and skip
+        IAddLine("");
+        return;
+    }
+
+    // only save history line if there is something there
+    if (fWorkingLine[0] != 0) {
+        // Save to history
+        strcpy(fHistory[fPythonMode].fData[fHistory[fPythonMode].fCursor], fWorkingLine);
```

Comment:
The instances of `strlen` were changed to `strnlen`. Is it worth similarly changing this instance of `strcpy` to `strncpy`?

Reply:
This is code that was copied and pasted from the `KEY_ENTER` case, and I would prefer to not change it in this PR.